core/sup_tree_core/periodic_log/message_builder.ex

```diff
@@ -0,0 +1,38 @@
+# Copyright(c) 2015-2019 ACCESS CO., LTD. All rights reserved.
+
+defmodule AntikytheraCore.PeriodicLog.MessageBuilder do
+  @max_proc_to_log     5
+  @max_msg_to_log      10
+  @queue_len_threshold 100
+
+  def init() do
+    nil
+  end
+
+  def build_log(state) do
+    procs =
+      :recon.proc_count(:message_queue_len, @max_proc_to_log)
+      |> Enum.filter(fn({_pid, qlen, _info}) -> qlen >= @queue_len_threshold end)
+    if procs != [] do
+      log_time = Antikythera.Time.to_iso_timestamp(Antikythera.Time.now())
+      log =
+        procs
+        |> Enum.reduce(log_time, fn({pid, qlen, info}, acc) ->
+          acc2 = acc <> "\n" <> Integer.to_string(qlen) <> " " <> inspect(info, structs: false)
+          process_info = Process.info(pid)
+          if process_info != nil do
+            process_info
+            |> Keyword.get(:messages)
+            |> Enum.take(@max_msg_to_log)
+            |> Enum.reduce(acc2, fn(msg, acc) -> acc <> "\n\t" <> inspect(msg, structs: false) end)
```

Comment:
I think nesting `Enum.reduce/3` (and using the same name `acc` for the accumulator) is a bit confusing.
How about separating it as a private function?

```elixir
defp add_process_info_to_log(partial_log, nil), do: partial_log
defp add_process_info_to_log(partial_log, process_info) do
  process_info
  |> Keyword.get(:messages)
  |> Enum.take(@max_msg_to_log)
  |> Enum.reduce(partial_log, fn(msg, acc) -> acc <> "\n\t" <> inspect(msg, structs: false) end)
end
```

Reply:
Extracted into separate functions. 0a6e8eef1604543b6f9e22b1d92b945aa1b50bdb